### PR TITLE
Switch CD to GitHub Actions

### LINKED ISSUE #159: Migrate release to GitHub Actions

### PR DESCRIPTION
**Pull request by simonbasle**
This commit switches the CD to GitHub Actions, with a few adjustments:

 - CI is now only run on pull requests (ci.yml)
 - Pushes to master trigger the `snapshot.yml` worflow which runs tests
 in one job then publishes to Artifactory /libs-snapshot-local in a
 second job
 - Publishing a GitHub Release or PreRelease triggers `release.yml`
   - the tag is used as the `version`, so it must end with .RELEASE or
   represent a pre-release and include .RC or .M (this is added as a
   condition to the publishing jobs)
   - tests are not run, as it is assumed that snapshot tests have been
   verified by the maintainer who triggers the release
   - prereleases go to Artifactory /libs-milestone-local
   - GA releases still go to Bintray for now

Segregation of deployments into two jobs allow to use an environment and
thus properly gate access to secrets. It also gives an opportunity to
avoid all 3rd party actions in the critical parts of the workflow.

Fixes #159.

### CODE REVIEW

**Comment by simonbasle** [comment]
Some caveats of the previous CI/CD and release process listed while working on this PR:

 - the `version` is NOT defined in gradle build files but rather defined at CD time in the script(s)
 - tags are made without a `v` prefix (important, as they're used as the `version` for a given release)
 - doesn't follow the calver scheme (important, as `.RELEASE` suffix is used to determine a release vs a milestone vs a snapshot)
 - `publish` doesn't trigger tests
 - tests must be run with JDK10+ due to examples
 - snapshoting on azure pipelines run with JDK11 configured for Gradle, but the target and source versions are still Java 8 anyway

File: .github/workflows/snapshot.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+jobs:
+  checkSnapshot:
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v1
+      - name: Set up JDK
+        uses: actions/setup-java@v1
+        with:
+          java-version: 13 #10+ is needed for Examples
```
Comment (bsideup) [inline .github/workflows/snapshot.yml:16]:
```suggestion
          java-version: 13 # Modern JVM is needed for Java 10+ specific tests
```

**Review by bsideup — COMMENTED** [review]
added a minor clarification

**Review by violetagg — APPROVED** [review]

**Review by bsideup — APPROVED** [review]

**Comment by simonbasle** [comment]
The `snapshot` environment has been prepared, so merging this should trigger a snapshot deployment.

**Pull request merged**